src/main/java/moze_intel/projecte/gameObjs/items/rings/SWRG.java

```diff
@@ -117,7 +119,11 @@ else if (stack.getTagCompound().getInteger(TAG_MODE) == 3)
 			toRemove = 0.64F;
 		}
 
-		removeEmc(stack, toRemove);
+		unprocessedEMC += toRemove;
+		if (unprocessedEMC > 1) {
+			removeEmc(stack, 1);
+			unprocessedEMC--;
+		}
```

Comment:
mutable state in an item is a huge no.

Reply:
same for every other item you do this

Reply:
It isn't static so it should be per item? Or is it because it is in Item, and that is on a lower tier than ItemStack, so they are all really using the same one.

Reply:
that's not how it works. every different itemstack of a swrg will share the same SWRG `Item` instance

Reply:
you probably need to put this in NBT

Reply:
Sounds good, I will take a look at it later today. I had forgotten that there is only one `Item` instance and that it wasn't like ItemStack where there are plenty of them.

Reply:
Fixed this, I also went ahead and made it so that it also stores the relay bonus, and partial EMC in collectors as well in NBT. Given initially I removed it for consistency and a number `< 1` was not really a big deal. I crossed out the previous text in the pull description.